lib/verifier/catalog_index_verifier.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "faraday"
+require_relative "../verifier"
+require_relative "../derivatives"
+
+# Verifies that catalog indexing workflow stage did what it was supposed to.
+
+module PostZephirProcessing
+  class CatalogIndexVerifier < Verifier
+    def verify_index_count(path:)
+      filename = File.basename(path)
+      if (m = filename.match(/^zephir_upd_(\d+)\.json\.gz/))
+        # in normal operation, we _should_ have indexed this the day after the
+        # date listed in the file.
+        #
+        # could potentially use the journal to determine when we actually
+        # indexed it?
+
+        date_of_indexing = Date.parse(m[1]) + 1
+        catalog_linecount = gzip_linecount(path: path)
+        solr_count = solr_count(date_of_indexing)
+      elsif /^zephir_full_\d+_vufind\.json\.gz/.match?(filename)
+        catalog_linecount = gzip_linecount(path: path)
+        solr_count = solr_nondeleted_records
+      else
+        raise ArgumentError, "#{path} doesn't seem to be a catalog index file"
+      end
+
+      if solr_count < catalog_linecount
+        error(message: "#{filename} had #{catalog_linecount} records, but only #{solr_count} had time_of_indexing on #{date_of_indexing} in solr")
+      end
+    end
+
+    def solr_count(date_of_indexing)
+      # get:
+      datebegin = date_of_indexing.to_datetime.new_offset(0).strftime("%FT%TZ")
+      dateend = (date_of_indexing + 1).to_datetime.new_offset(0).strftime("%FT%TZ")
+      solr_result_count("time_of_index:#{datebegin}%20TO%20#{dateend}]")
```

Comment:
We can probably simplify this and make it more robust by just checking that there are at least N items that were indexed at some point after the file was produced. Not foolproof, but mitigates against the case of when we index N items on the 1st and then those same N again on the 2nd (updating their last index date).